File: 5th_Algo/이민규/BJ_2745.c
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    char input[10] = { 0 };
    int base = 0;
    int result = 0;
    int temp = 0;
    scanf("%s %d", input, &base);
    int len = strlen(input);

    for (int i = 0; i < len; i++)
    {
        if (input[i] >= 'A' && input[i] <= 'Z')
            temp = input[i] - 'A' + 10;
        else
            temp = input[i] - '0';

        result += temp * pow(base, len - i - 1);
    }
    printf("%d", result);

    return 0;
}

File: 5th_Algo/이민규/BJ_2745.py
input_data = input().split()
string = input_data[0]
base = int(input_data[1])
temp = 0
result = 0

for i in range(len(string)):
    c = string[i]
    if c >= 'A' and c <= 'Z':
        temp = ord(c) - ord('A') + 10
    else:
        temp = ord(c) - ord('0')
    
    result = result + temp * pow(base, len(string) - i - 1)
    
print(result)
